RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -0,0 +1,317 @@
+    }
+
+    const auto &jet_ref = tag_infos->at(jet_n).jet();
+    for (std::size_t flav_n = 0; flav_n < flav_names_.size(); flav_n++) {
+      (*(output_tags[flav_n]))[jet_ref] = outputs[flav_n];
+    }
+  }
+
+  if (debug_) {
+    std::cout << "=== " << iEvent.id().run() << ":" << iEvent.id().luminosityBlock() << ":" << iEvent.id().event()
+              << " ===" << std::endl;
+    for (unsigned jet_n = 0; jet_n < tag_infos->size(); ++jet_n) {
+      const auto &jet_ref = tag_infos->at(jet_n).jet();
+      std::cout << " - Jet #" << jet_n << ", pt=" << jet_ref->pt() << ", eta=" << jet_ref->eta()
+                << ", phi=" << jet_ref->phi() << std::endl;
+      for (std::size_t flav_n = 0; flav_n < flav_names_.size(); ++flav_n) {
+        std::cout << "    " << flav_names_.at(flav_n) << " = " << (*(output_tags.at(flav_n)))[jet_ref] << std::endl;
+      }
+    }
+  }
+
+  // put into the event
+  for (std::size_t flav_n = 0; flav_n < flav_names_.size(); ++flav_n) {
+    iEvent.put(std::move(output_tags[flav_n]), flav_names_[flav_n]);
+  }
+}
+
+std::vector<float> ParticleNetSonicJetTagsProducer::center_norm_pad(const std::vector<float> &input,
+                                                                    float center,
+                                                                    float norm_factor,
+                                                                    unsigned min_length,
+                                                                    unsigned max_length,
+                                                                    float pad_value,
+                                                                    float replace_inf_value,
+                                                                    float min,
+                                                                    float max) {
+  // do variable shifting/scaling/padding/clipping in one go
+
+  assert(min <= pad_value && pad_value <= max);
+  assert(min_length <= max_length);
+
+  unsigned target_length = std::clamp((unsigned)input.size(), min_length, max_length);
+  std::vector<float> out(target_length, pad_value);
+  for (unsigned i = 0; i < input.size() && i < target_length; ++i) {
+    out[i] = std::clamp((catch_infs(input[i], replace_inf_value) - center) * norm_factor, min, max);
+  }
+  return out;
+}
+
+void ParticleNetSonicJetTagsProducer::acquire(edm::Event const &iEvent, edm::EventSetup const &iSetup, Input &iInput) {
+  client_->setBatchSize(batchSize_);
+  edm::Handle<TagInfoCollection> tag_infos;
+  iEvent.getByToken(src_, tag_infos);
+  if (!tag_infos->empty()) {
+    unsigned jet_n = 0;
+    const auto &taginfo = (*tag_infos)[jet_n];
+    for (unsigned igroup = 0; igroup < input_names_.size(); ++igroup) {
+      const auto &group_name = input_names_[igroup];
+      auto &input = iInput.at(group_name);
+      auto tdata = std::make_shared<TritonInput<float>>(1);
```

Comment:
this can be rebased to a newer release to use `allocate<float>()` (from https://github.com/cms-sw/cmssw/pull/33801)

Reply:
I did this per our discussion on slack.  "allocate" function now used here